project/controller/newRouteController.php

```diff
@@ -1,58 +1,73 @@
 <?php
 
 include_once "../constants/routeConstants.php";
-
 include_once '../utils/sessionUtils.php';
 include_once '../database/routeUtils.php';
 
+loginRequired();
+
+
 $distance= getDataFromPOST('distance');
 $name = getDataFromPOST('routeName');
 $mode = getDataFromPOST('mode');
-$team = getDataFromPOST('teamID');
+$team = getDataFromPOST('team');
 $startLatitude = getDataFromPOST('startLatitude');
 $startLongitude = getDataFromPOST('startLongitude');
 $endLatitude = getDataFromPOST('endLatitude');
 $endLongitude = getDataFromPOST('endLongitude');
 
 $userId = getActiveUserID();
 
-if($userId === null) {
-    //TODO: Session is not initialized, move to login?
-}
-
-if($mode === null) {
-    //TODO: Is this check even needed? Can user somehow hack POST so he does not post one value?
+if (!nullCheck(array($distance, $name, $mode, $team, $startLatitude, $startLongitude, $endLatitude, $endLongitude))) {
```

Comment:
Team can be null, when saving public route. Or private route

Reply:
oh youre right, my bad